### ISSUE TITLE
What is pynvshmem?

### ISSUE THREAD

**Issue by Willy-Chan** [ByteDance-Seed/Triton-distributed#140]
New here.

In the tutorial docs pynvshmem seems to come from triton distributed:
```python
import torch
import triton
import triton.language as tl
from triton_dist.language.extra import libshmem_device
from triton_dist import pynvshmem
```

But in the examples in the repo it looks like it comes from NVSHMEM4Py. Could I get clarification on what this is exactly?

Also in the tutorial [here](https://triton-distributed.readthedocs.io/en/latest/getting-started/tutorials/02-intra-node-allgather.html#sphx-glr-getting-started-tutorials-02-intra-node-allgather-rst), there are functions like `pynvshmem.nvshmem_create_tensor_list_intra_node([M, N], dtype)`, but these are nowhere to be found in NVSHMEM4Py. Are the docs outdated?
